src/opm/io/eclipse/EGrid.cpp

```diff
@@ -28,43 +30,188 @@
+            } else {
                act_index.push_back(-1);
-           }
+            }
         }
-   } else {
-       int nCells = nijk[0] * nijk[1] * nijk[2];
-       act_index.resize(nCells);
-       glob_index.resize(nCells);
-       std::iota(act_index.begin(), act_index.end(), 0);
-       std::iota(glob_index.begin(), glob_index.end(), 0);
-   }
-
-   coord_array = get<float>("COORD");
-   zcorn_array = get<float>("ZCORN");
+    } else {
+        int nCells = nijk[0] * nijk[1] * nijk[2];
+        act_index.resize(nCells);
+        glob_index.resize(nCells);
+        std::iota(act_index.begin(), act_index.end(), 0);
+        std::iota(glob_index.begin(), glob_index.end(), 0);
+    }
+
+    if (hostnum_index > -1){
+        auto hostnum = getImpl(hostnum_index, INTE, inte_array, "integer");
+        host_cells.reserve(hostnum.size());
+
+        for (auto val : hostnum)
+            host_cells.push_back(val -1);
+    }
 }
 
+std::vector<std::array<int, 3>> EGrid::hostCellsIJK()
+{
+    std::vector<std::array<int, 3>> res_vect;
+    res_vect.reserve(host_cells.size());
+
+    for (auto val : host_cells){
+        std::array<int, 3> tmp;
+        tmp[2] = val / (host_nijk[0] * host_nijk[1]);
+        int rest = val % (host_nijk[0] * host_nijk[1]);
+
+        tmp[1] = rest / host_nijk[0];
+        tmp[0] = rest % host_nijk[0];
+
+        res_vect.push_back(tmp);
+    }
+
+    return res_vect;
+}
+
+std::vector<NNCentry> EGrid::get_nnc_ijk() const
+{
+    if (!m_nncs_loaded)
+        load_nnc_data();
+
+    std::vector<NNCentry> res_vect;
+    res_vect.reserve(nnc1_array.size());
+
+    for (size_t n=0; n< nnc1_array.size(); n++){
+        auto ijk1 = ijk_from_global_index(nnc1_array[n] - 1);
+        auto ijk2 = ijk_from_global_index(nnc2_array[n] - 1);
+
+        if (transnnc_array.size() > 0)
+            res_vect.push_back({ijk1[0], ijk1[1], ijk1[2], ijk2[0], ijk2[1], ijk2[2], transnnc_array[n]});
+        else
+            res_vect.push_back({ijk1[0], ijk1[1], ijk1[2], ijk2[0], ijk2[1], ijk2[2], -1.0 });
+    }
+
+    return res_vect;
+}
+
+
+void EGrid::load_grid_data() const
+{
+    coord_array = getImpl(coord_array_index, REAL, real_array, "float");
+    zcorn_array = getImpl(zcorn_array_index, REAL, real_array, "float");
+}
+
+void EGrid::load_nnc_data() const
+{
+    if ((nnc1_array_index > -1) && (nnc2_array_index > -1)) {
+
+        nnc1_array = getImpl(nnc1_array_index, Opm::EclIO::INTE, inte_array, "inte");
+        nnc2_array = getImpl(nnc2_array_index, Opm::EclIO::INTE, inte_array, "inte");
+
+        if ((Opm::filesystem::exists(initFileName)) && (nnc1_array.size() > 0)){
+            Opm::EclIO::EInit init(initFileName);
+
+            auto init_dims = init.grid_dimension(m_grid_name);
+            int init_nactive = init.activeCells(m_grid_name);
+
+            if ((init_dims != nijk) or (init_nactive != nactive)){
+                OPM_THROW(std::invalid_argument, "inconsistent grid dimension in init file");
```

Comment:
found blah, expected blah. also 'or' is for wusses ;p

Reply:
Throw message improved. Also split original if statement (including or) into two different if statements. 

NNC definition and NNC transmissibilities store in different files (egrid and init file). I would really like to be sure that these are compatible.